### PR TITLE
CRM-17080 - Smarty api cleanup

### PR DESCRIPTION
**Pull request by colemanw**
https://issues.civicrm.org/jira/browse/CRM-17080

### CODE REVIEW

File: templates/CRM/Admin/Page/APIExplorer.tpl [hunk 1]
```diff
@@ -162,7 +162,7 @@
       <label for="api-entity">{ts}Entity{/ts}:</label>
       <select class="crm-form-select big required" id="api-entity" name="entity">
         <option value="" selected="selected">{ts}Choose{/ts}...</option>
-        {crmAPI entity="Entity" action="get" var="entities" version=3}
+        {crmAPI entity="Entity" var="entities"}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/APIExplorer.tpl]: Doesn't this mean the api explorer will fail to show action for non-get actions too - where it should be provided?

Comment (colemanw) [inline templates/CRM/Admin/Page/APIExplorer.tpl]: No, it means the smarty_function_crmAPI defaults to "get" so there's no need to explicitly pass it in. Removed `version=3` for the same reason.

Comment (eileenmcnaughton) [inline templates/CRM/Admin/Page/APIExplorer.tpl]: right - but isn't this tpl providing sample code? Including for non-get actions?

Comment (colemanw) [inline templates/CRM/Admin/Page/APIExplorer.tpl]: Line 165 is populating the "Entity" selector in the api explorer; it's not sample code - that's provided elsewhere.

Comment (colemanw) [inline templates/CRM/Admin/Page/APIExplorer.tpl]: If you're asking if this code changes the behavior of the smarty api or the api explorer, no it doesn't, it's just code cleanup. The biggest change is to unset the correct variables in smarty_function_crmAPI (the variables are named `var` and `action` but previously it was unsetting nonexistant variables `assign` and `method`). Nothing else is changed. The api explorer will continue to put `action` in its generated code (even though that's redundant if the action is "get" but who cares).

**Comment by eileenmcnaughton** [comment]
let's merge this puppy

**Pull request merged**